k8s/operators/src/diskpool/crd.rs

```diff
@@ -67,22 +71,39 @@ pub enum PoolState {
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq, JsonSchema)]
-/// Status of the pool which is driven and changed by the controller loop
+#[non_exhaustive]
+/// PoolStatus is Control plane status of a given DSP CR.
+pub enum PoolStatus {
+    /// State is Unknown.
+    Unknown,
+    /// The pool is in normal working order.
+    Online,
+    /// The pool has experienced a failure but can still function.
+    Degraded,
+    /// The pool is completely inaccessible.
+    Faulted,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq, JsonSchema)]
+/// Status of the pool which is driven and changed by the controller loop.
 pub struct DiskPoolStatus {
-    /// The state of the pool
+    /// The state of the pool.
     pub state: PoolState,
-    /// Capacity as number of bytes
+    /// Pool status from respective control plane object.
+    pub status: Option<PoolStatus>,
```

Comment:
See my earlier comment, state and status mean the same thing, so a more descriptive name for this variable will make it easier for developers.

Reply:
Yeah, the variable names can be `control_plane_status` and `cr_status` respectively for clarity. I agree.

Reply:
What about `cr_status` and `pool_status`?

Reply:
Sounds good.

Reply:
I agree. We need to make names more clear.

Reply:
I thought the suggestion was pool_status?
Using "state " and "status" to distinguish between different things arbitrarily isn't good IMHO.

Reply:
Yes, made it pool_status. This is control plane status for the dsp.